text/0010-bft-signatures.md

```diff
@@ -0,0 +1,291 @@
+even if the block itself contains only a single transaction (that is often 3.5KB in size itself).
+
+Clearly, encoding identities each time in the signature header of each orderer signature is wasteful and therefore we should strive to eliminate this overhead.
+
+## Existing approaches for block replication and block signature verification
+[Existing approaches]: #existing-approaches-for-block-replication-and-block-signature-verification
+
+### Attestation for block existence
+
+An approach for block denial of service resistance taken by [SmartBFT](https://arxiv.org/abs/2107.06922) is to connect to (at least)`f+1` ordering service nodes, fetch blocks only from one of them and fetch attestations that blocks exist from the remaining `f` (or more).
+If a block existence attestation for a sequence **i** is found valid but the orderer designated to send blocks withholds block **i** for too long, the role assignment of block source and attestation for block existence source is re-shuffled.
+
+An attestation for block existence is simply a regular block (along with its `2f+1` signatures) pruned from its transactions. The signatures can still be verified despite the transactions being pruned from the block thanks
+to the fact that in Fabric the signing algorithm is applied on the ASN1 encoding of the header block which carries the hash of the concatenation of all transactions:
+
+```
+message BlockHeader {
+    uint64 number = 1; // The position in the blockchain
+    bytes previous_hash = 2; // The hash of the previous block header
+    bytes data_hash = 3; // The hash of the BlockData
+}
+```
+
+Let's consider the two possible outcomes based on the orderer that is selected for block retrieval:
+
+1. If orderer that blocks are pulled from is withholding newly formed blocks, then there is at least one honest orderer that will not withhold attestations of block existence. Hence, a re-shuffling of role assignment will occur until an honest orderer will be the block source.
+2. Else, the orderer that blocks are pulled from is not withholding newly formed blocks. However, it is possible that new blocks are not formed due to lack of client activity. It is imperative to ensure that malicious ordering nodes cannot make honest ordering nodes be falsely suspected of block withholding.
+As each "block existence attestation" contains `2f+1` signatures it is impossible to forge one. Hence, if a node sends a attestation for block existence, it means this a block with such sequence number was indeed appended to the blockchain.
+
+In order for a peer or an orderer to convey the intent of receiving only blocks without transactions, 
+the [SeekInfo](https://github.com/hyperledger/fabric-protos/blob/main/orderer/ab.proto#L45) message is expanded to contain also a `SeekContentType` field:
+
+```
+  // SeekContentType indicates what type of content to deliver in response to a request. If BLOCK is specified,
```

Comment:
You can add an API in the block storage regardless, and just call it in the deliver server. 
I don't see why it's a problem that a peer will also be able to service this API. 
After all, it's the same common deliver server code in both node types.

Reply:
May be you did not get the main suggestion here... let me paste again 

Given the current pattern of deliver service interfaces in [order](https://github.com/hyperledger/fabric-protos/blob/main/orderer/ab.proto) and [peer](https://github.com/hyperledger/fabric-protos/blob/6e299ac721985b2f616065aca4d0cb63dffeca40/peer/events.proto#L65), it makes better sense to define a separate service instead of fitting in `SeekInfo`

Reply:
I defined a separate gRPC service as you suggested.